ludwig/api.py

```diff
@@ -891,6 +910,46 @@ def _tune_batch_size(self, trainer, dataset, random_seed: int = default_random_s
         trainer.eval_batch_size = self.config_obj.trainer.eval_batch_size
         trainer.gradient_accumulation_steps = self.config_obj.trainer.gradient_accumulation_steps
 
+    def generate(
+        self,
+        input_strings: Union[str, List[str]],
+        generation_config: Optional[dict] = None,
+    ) -> Union[str, List[str]]:
+        """A simple generate() method that directly uses the underlying transformers library to generate text."""
+        if self.config_obj.model_type != MODEL_LLM:
+            raise ValueError(
+                f"Model type {self.config_obj.model_type} is not supported by this method. Only `llm` model type is "
+                "supported."
+            )
+        if not torch.cuda.is_available():
+            raise ValueError("GPU is not available.")
```

Comment:
is this required?

Reply:
It seems to be for loading quantized models: https://github.com/ludwig-ai/ludwig/issues/3695. This error string might be preferable to the other one. WDYT?

Reply:
Nice! Yes I agree, thanks for the clarification. Can you put a comment there linking to issue 3695?